### PR TITLE
FeatureLinePointsClass

### PR DESCRIPTION
**Pull request by BartDeMaesschalckAtDIMCO**
I have added a FeatueLinePoints class. This is used to export featureline-points into a list from a bunch of featurelines with a given code. (Baseline.GetFeaturelinePointsByCode(string code))
In our workflow we do not have much use for the featurelines themselves because we have to create execution models and a such (on linear infra projects) do not have much use for nicely curve walls, floors or roofs because we place them in straight segments.
The segments are calculated using a corridor with a fixed frequency and in the export the geometry points of alignments and profiles are ignored.

The extra class bundles every importing thing known for that point:
- position (optional transformed)
- coordinate system on the alignment in that point
- code of the point (from featureline)

### CODE REVIEW

File: Src/CivilConnection/2020/Autodesk2020/Baseline.cs [hunk 1]
```diff
@@ -1019,6 +1078,131 @@ public IList<IList<Featureline>> GetFeaturelinesByCode(string code)
...
+                {
+                    List<FeaturelinePoint> points = new List<FeaturelinePoint>();
+
+                    foreach (var pt in fl.FeatureLinePoints.Cast<AeccFeatureLinePoint>())
+                    {
+                        var p = pt.XYZ;
+
+                        try
+                        {
+                            Point _pt = Point.ByCoordinates(p[0], p[1], p[2]).Transform(_totalTransform);
```
Comment (paoloemilioserra) [inline Src/CivilConnection/2020/Autodesk2020/Baseline.cs:1127]: need to cast to Point

File: Src/CivilConnection/2020/Autodesk2020/Baseline.cs [hunk 2]
```diff
@@ -1019,6 +1078,131 @@ public IList<IList<Featureline>> GetFeaturelinesByCode(string code)
...
+                            foreach (var fl in fs.Cast<AeccFeatureLine>())
+                            {
+                                List<FeaturelinePoint> points = new List<FeaturelinePoint>();
+
+                                foreach (var pt in fl.FeatureLinePoints.Cast<AeccFeatureLinePoint>())
+                                {
+                                    var p = pt.XYZ;
+                                    try
+                                    {
+                                        Point _pt = Point.ByCoordinates(p[0], p[1], p[2]).Transform(_totalTransform);
```
Comment (paoloemilioserra) [inline Src/CivilConnection/2020/Autodesk2020/Baseline.cs:1170]: need to cast to Point

File: Src/CivilConnection/2020/Autodesk2020/Baseline.cs [hunk 3]
```diff
@@ -1019,6 +1078,131 @@ public IList<IList<Featureline>> GetFeaturelinesByCode(string code)
...
+                                        //calculating the station of the resulting easting, northing on the main aligment.
+                                        //b.Alignment.StationOffset(_x, _y, out _s, out _o);
+                                        points.Add(/*Math.Round(_s, 5),*/ Point.ByCoordinates(p[0], p[1], p[2]));
+                                    }
+                                    catch (Exception ex)
+                                    {
+                                        Utils.Log(string.Format("ERROR: {0}", ex.Message));
+                                    }
+                                }
+                                blFeaturelinesPoints.Add(Point.PruneDuplicates(points).ToList());
```
Comment (paoloemilioserra) [inline Src/CivilConnection/2020/Autodesk2020/Baseline.cs:1186]: Could be slow

**Review by paoloemilioserra — COMMENTED** [review]
It is best to consider the FeaturelinePoints as a property of the Featureline object rather than extracting it twice.

**Comment by BartDeMaesschalckAtDIMCO** [comment]
Some old code was somehow pushed...

It is best to consider the FeaturelinePoints as a property of the Featureline object rather than extracting it twice -> Whar doe you means by extracting it twice?

Regards,

Bart

**Pull request closed without merging**